### ISSUE TITLE
choice of next instance not deterministic

### ISSUE THREAD

**Issue by mfeurer** [automl/SMAC3#62]
I just found the following code snippet in intensification.py. Instead of using the instance of the random number generator, it uses a global instance of random:

```
if available_insts:
    # Line 5 (here for easier code)
    next_instance = random.choice(list(available_insts))
```

**Comment by mlindauer** [comment]
Do you want to fix it in a new PR?
Or do we wan to fix it in #60?

**Comment by mfeurer** [comment]
I think it should not be in #60 because it solves a different issue. Unrelated note on #60: I think the PR is too big and it tries to solve too many issues at the same time. If it would only add a nice facade to SMAC, fixing something in intensification.py wouldn't create any merge conflicts.
